### PR TITLE
resolved_ts: track ingest sst

### PR DESCRIPTION
**Pull request by hehechen**
Signed-off-by: hehechen <awd123456sss@gmail.com>

### What is changed and how it works?

Issue Number: Ref #13353

What's Changed:

```
Update tracked_index in `CmdType::IngestSst` scenario.
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Manual test (add detailed scripts or steps below)
Add log in TiFlash proxy to see whether tracked_index is updated.

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/resolved_ts/src/cmd.rs [hunk 1]
```diff
@@ -33,6 +33,7 @@ pub enum ChangeRow {
         commit_ts: TimeStamp,
         value: Option<Value>,
     },
+    IngestSsT {},
```
Comment (5kbpers) [inline components/resolved_ts/src/cmd.rs:36]:
```suggestion
    IngestSsT,
```

File: components/resolved_ts/src/cmd.rs [hunk 2]
```diff
@@ -58,8 +59,11 @@ impl ChangeLog {
                         let flags =
                             WriteBatchFlags::from_bits_truncate(request.get_header().get_flags());
                         let is_one_pc = flags.contains(WriteBatchFlags::ONE_PC);
-                        let changes = group_row_changes(request.requests.into());
-                        let rows = Self::encode_rows(changes, is_one_pc);
+                        let (changes, has_ingest_sst) = group_row_changes(request.requests.into());
+                        let mut rows = Self::encode_rows(changes, is_one_pc);
+                        if has_ingest_sst {
+                            rows.push(ChangeRow::IngestSsT {});
```
Comment (5kbpers) [inline components/resolved_ts/src/cmd.rs:65]:
```suggestion
                            rows.push(ChangeRow::IngestSsT);
```

**Review by 5kbpers — COMMENTED** [review]
Could you add an integration test (in `resolved_ts/tests`) for these changes?

**Comment by hehechen** [comment]
> 

done

**Comment by hehechen** [comment]
> Could you add an integration test (in `resolved_ts/tests`) for these changes?

Done.

**Review by 5kbpers — APPROVED** [review]

File: components/raftstore/src/store/util.rs [hunk 3]
```diff
@@ -931,6 +931,14 @@ impl RegionReadProgressRegistry {
             .map(|rp| rp.safe_ts())
     }
 
+    pub fn get_tracked_index(&self, region_id: &u64) -> Option<u64> {
+        if let Some(rp) = self.get(region_id) {
```
Comment (ethercflow) [inline components/raftstore/src/store/util.rs:935]:
```suggestion
        self.registry
```

File: components/raftstore/src/store/util.rs [hunk 4]
```diff
@@ -931,6 +931,14 @@ impl RegionReadProgressRegistry {
...
+            let core = rp.core.lock().unwrap();
```
Comment (ethercflow) [inline components/raftstore/src/store/util.rs:936]:
```suggestion
            .lock()
```

File: components/raftstore/src/store/util.rs [hunk 5]
```diff
@@ -931,6 +931,14 @@ impl RegionReadProgressRegistry {
...
+            return Some(core.applied_index);
```
Comment (ethercflow) [inline components/raftstore/src/store/util.rs:937]:
```suggestion
            .unwrap()
```

File: components/raftstore/src/store/util.rs [hunk 6]
```diff
@@ -931,6 +931,14 @@ impl RegionReadProgressRegistry {
...
+        }
```
Comment (ethercflow) [inline components/raftstore/src/store/util.rs:938]:
```suggestion
        .get(region_id)
```

File: components/raftstore/src/store/util.rs [hunk 7]
```diff
@@ -931,6 +931,14 @@ impl RegionReadProgressRegistry {
...
+        None
```
Comment (ethercflow) [inline components/raftstore/src/store/util.rs:939]:
```suggestion
        .map(|rp| rp.core.lock().unwrap().applied_index)
```

**Review by ethercflow — APPROVED** [review]
Rest LGTM

**Pull request merged**